test/parallel/test-http2-too-many-headers.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const http2 = require('http2');
+const assert = require('assert');
+const {
+  NGHTTP2_ENHANCE_YOUR_CALM
+} = http2.constants;
+
+// By default, the maximum number of header fields allowed per
+// block is 1024, including the HTTP pseudo-header fields, by
+// setting one here, no request should be acccepted because it
+// does not allow for even the basic HTTP pseudo-headers
+const server = http2.createServer({ maxHeaderListPairs: 1 });
```

Comment:
I think these types of settings should throw. What's the minimum?

Reply:
We can set whatever minimum we want here. For the request to work, we need at least 4 to account for the required pseudoheaders.

Reply:
I would say we validate it's greater than 4.

Reply:
how about rather than report an error, we just automatically allow for the number of pseudo-headers? So if a user sets 0, the internal value is 4....